collector/container.go

```diff
@@ -327,6 +286,71 @@ func (c *ContainerMetricsCollector) collect(ch chan<- prometheus.Metric) (*prome
 		)
 	}
 
+	hnsEndpoints, err := hcsshim.HNSListEndpointRequest()
+	if err != nil {
+		log.Warn("Failed to collect network stats for containers")
+		return nil, nil
+	}
+
+	if len(hnsEndpoints) == 0 {
+		log.Info("No network stats for containers to collect")
+		return nil, nil
+	}
+
+	for _, endpoint := range hnsEndpoints {
+		endpointStats, err := hcsshim.GetHNSEndpointStats(endpoint.Id)
+		if err != nil {
+			log.Warn("Failed to collect network stats for interface %s", endpoint.Id)
+			continue
+		}
+
+		for _, containerId := range endpoint.SharedContainers {
+			containerPrefix, ok := containerPrefixes[containerId]
+			endpointId := strings.ToUpper(endpoint.Id)
+			if ok {
+				ch <- prometheus.MustNewConstMetric(
+					c.BytesReceived,
+					prometheus.CounterValue,
+					float64(endpointStats.BytesReceived),
+					containerPrefix, endpointId,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.BytesSent,
+					prometheus.CounterValue,
+					float64(endpointStats.BytesSent),
+					containerPrefix, endpointId,
+				)
+				ch <- prometheus.MustNewConstMetric(
+					c.PacketsReceived,
+					prometheus.CounterValue,
+					float64(endpointStats.PacketsReceived),
+					containerPrefix, endpointId,
+				)
+				ch <- prometheus.MustNewConstMetric(
+					c.PacketsSent,
+					prometheus.CounterValue,
+					float64(endpointStats.PacketsSent),
+					containerPrefix, endpointId,
+				)
+				ch <- prometheus.MustNewConstMetric(
+					c.DroppedPacketsIncoming,
+					prometheus.CounterValue,
+					float64(endpointStats.DroppedPacketsIncoming),
+					containerPrefix, endpointId,
+				)
+				ch <- prometheus.MustNewConstMetric(
+					c.DroppedPacketsOutgoing,
+					prometheus.CounterValue,
+					float64(endpointStats.DroppedPacketsOutgoing),
+					containerPrefix, endpointId,
+				)
+			} else {
+				log.Warn("Failed to collect network stats for container %s", containerId)
```

Comment:
For readability, I might reverse this check to be something like:

```
if !ok {
log.Warn("Failed to collect network stats for container %s", containerId)
continue
}

do all metric stuff....
```

Also something to consider is did this failed to collect or did we get a network stat that we might not have a contianer for (due to a timing issue?) this message is probably ok but I think its a slight distinction between the two

Reply:
### Suggestion 1
> For readability, I might reverse this check to be something like...

Refactored as suggested for better readability.

### Suggestion 2
> ...this message is probably ok but I think its a slight distinction between the two

This particular error would only occur if as you mentioned, we get a network stat that we do not have a container for. Ideally we could just fetch network endpoints and not have to match with a container id and avoid this potential scenario, but then this would not be backwards compatible with the old manner of specifying the container id and container runtime as labels for the metric. Either way, this scenario for failure should not really occur but it could due to a timing issue or something else. Am happy to change the error message if you have a better suggestion?

Reply:
The changes lgtm, easier to read!

Reply:
as for the error message, with the change it seems to reads better to me and I think we are in good shape